src/main.rs

```diff
@@ -93,7 +95,10 @@ async fn main() {
         port as u16,
     );
 
-    let app = Router::new()
+    let mut headers = HeaderMap::new();
```

Comment:
The implementation is correct but I've already created a layer `default_headers` . Could you add it there? I think it is better not to do such low level stuff in main function and keep in the `middleware` module.
Alternatively, use tower_default_headers::DefaultHeadersLayer but move the content of `default_headers` there as well, but in the `middleware.rs`

Reply:
I added it in default_headers function